tools/swapnode/swapnode.cpp

```diff
@@ -268,24 +268,34 @@ int main(int argc, const char *argv[])
                     case cmd_addspares:
                     case cmd_removespares:
                     {
+                        std::vector<std::string> hosts;
                         SocketEndpointArray allEps;
                         unsigned p=2;
                         do
                         {
-                            const char *ipOrRange = params.item(p);
+                            const char *hostOrIpRange = params.item(p);
                             SocketEndpointArray epa;
-                            epa.fromText(ipOrRange, 0);
-                            ForEachItemIn(e, epa)
-                                allEps.append(epa.item(e));
+                            epa.fromText(hostOrIpRange, 0);
+                            if (epa.ordinality()>1)
+                            {
+                                ForEachItemIn(e, epa)
+                                {
+                                    StringBuffer ipStr;
+                                    epa.item(e).getIpText(ipStr);
```

Comment:
Won't this always return an IP ?  Did you want to preserve the name if it was that ??

Reply:
yes, but this is if it's detected an IP range (e.g. 192.168.11-15) in the parameters - so it's explicitly dealing with IP's at this point.
If not a range (the else block a few lines below), it doesn't touch it, i.e. it leaves it as a host if that's what it was.